webui/react/src/e2e/auth.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { expect } from '@playwright/test';
+
+import { test } from './global-fixtures';
+
+test.describe('Authentication', () => {
+  const USERNAME = process.env.PW_USER_NAME ?? '';
+  const PASSWORD = process.env.PW_PASSWORD ?? '';
+
+  test.beforeEach(async ({ dev }) => {
+    await dev.setServerAddress();
+  });
+
+  test('Login and Logout', async ({ page }) => {
+    await page.goto('/');
+
+    await test.step('Login steps', async () => {
+      await page.getByPlaceholder('username').fill(USERNAME);
+      await page.getByPlaceholder('password').fill(PASSWORD);
+      await page.getByRole('button', { name: 'Sign In' }).click();
```

Comment:
consider abstracting the login process to the fixture as a page object.

Reply:
fixed